### ISSUE TITLE
Navigation: Problem rotating 3Dview with 3D-mouse (jumps when switching)

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#6214]
Issue imported from https://tracker.freecad.org/view.php?id=4654

* **Reporter:** SeppoTu
* **Date submitted:** 5/4/2021
* **FreeCAD version:** 0.19
* **Category:** Bug
* **Status:** new
* **Tags:** 3dconnexion, mouse, rotate, Win

# Original report text

Center of rotation is not updated in perspective view mode, during transfer with 3D-mouse.
In perspective view mode the normal mouse operations sets rotating center in correct postion.
Using 3D-mouse the rotation center is not updated. This makes it very difficult to use 3D-mouse in prespective view mode.
Tested with 3DConnexion's SpacePilot Pro.

Forum thread: https://forum.freecadweb.org/viewtopic.php?f=3&t=67282

# Steps to reproduce

1. Open FreeCAD.
2. Create new part (cube).
4. Change view mode to perspective view.
5. Rotate cube with 3D-mouse. Rotation center is in cube's center.
6. ~Move cube away from it's original position by using 3D-mouse.~ Zoom the view with 3D-mouse, not that cube it self has to move 
7. Rotate cube again with 3D-mouse. Rotation center has not transferred with the cube.
   Instead rotation center is still at the old location.
8. Rotate cube with normal mouse. Cube jumps to to it's original location (rotation center) before it starts rotate.

# FreeCAD Info

```
OS: Windows 10 Version 2009
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.19.24276 (Git)
Build type: Release
Branch: releases/FreeCAD-0-19
Hash: a88db11e0a908f6e38f92bfc5187b13ebe470438
Python version: 3.8.6+
Qt version: 5.15.1
Coin version: 4.0.1
OCC version: 7.5.0
Locale: Finnish/Finland (fi_FI)

```

# Other bug information

* **Priority:** normal
* **Severity:** minor
* **Category:** Bug
* **OS: Windows 10 Pro 20H2**
* **Platform:** Table PC
* **Updated:** 5/5/2021

# Discussion from Mantis ticket

### Comment by SeppoTu 2021-05-05 07:45
Correction to reproduction:
In reproduce step 6, I ment to say that: zoom the view with 3D-mouse, not that cube it self has to move.

**Comment by adrianinsaval** [comment]
is this still relevant? anybody else with a 3d mouse can check this?

**Comment by luzpaz** [comment]
@chennes, you do right ?

**Comment by chennes** [comment]
Yeah - I don't think I am reproducing this exact behavior, but I definitely get some rotation glitches in perspective view, with sudden jumps between angles. Maybe I'm just garbage at using the mouse (I am still new at it), but it doesn't seem right to me.

**Comment by FEA-eng** [comment]
This seems to be fixed by #12929 but it would be nice if someone could confirm.

**Comment by chennes** [comment]
I just checked, and there is still some kind of problem switching from SpaceMouse to normal mouse.  If I follow the above instructions through step 6, it appears that the rotation center indicator DOES move with the cube. But when I switch to the normal mouse, the cube still jumps in position (a zoom, perhaps?) as soon as I begin a rotation.

**Comment by luzpaz** [comment]
Can someone re-test this ?

**Comment by FEA-eng** [comment]
> Can someone re-test this ?

What Chris mentioned (jump when switching from 3D mouse to regular mouse) is still relevant.

![3d mouse jump](https://github.com/user-attachments/assets/cbee45a8-f2b4-4b1f-ae2a-c621b0e231c9)

**Comment by luzpaz** [comment]
@oficinerobotica can you test to see if this is still relevant on your 3dconnexion mouse ?

**Comment by FEA-eng** [comment]
I don't get the jump when switching to a normal mouse now. But the part is shaking strangely (it doesn't happen in 1.0):

![Image](https://github.com/user-attachments/assets/711b956d-8d52-4c4e-9519-4480c861fac7)

```
OS: Windows 10 build 19045
Architecture: x86_64
Version: 1.1.0dev.39980 (Git) Conda
Build type: Release
Branch: main
Hash: 836ad91f3ddd3ec63b84b8e3f87a767d5465261d
Python 3.11.11, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Polish/Poland (pl_PL)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
```

**Comment by oficinerobotica** [comment]
I can't seem to reproduce the jumping nor the shaking but I tested on Linux atm, and the space mouse driver is limited respect to the windows version.  @FEA-eng, On the spacemouse side,  do you  have "lock horizon" enabled by chance? there are so many variables envolved that could be tested to try to narrow down the problem

https://github.com/user-attachments/assets/7e846e5a-3008-491a-93aa-a45495d635fc

**Comment by FEA-eng** [comment]
> I can't seem to reproduce the jumping nor the shaking but I tested on Linux atm, and the space mouse driver is limited respect to the windows version. [@FEA-eng](https://github.com/FEA-eng), On the spacemouse side, do you have "lock horizon" enabled by chance? there are so many variables envolved that could be tested to try to narrow down the problem

Sorry for the late reply, I missed your comment. I changed my computer but if "Lock horizon" wasn't enabled by default, I had it off because I didn't change such settings.

Now I retried with new weekly and I don't see the aforementioned issues anymore. @chennes Can you check too? Perhaps this issue should be closed now.

**Comment by FEA-eng** [comment]
Needs feedback from @chennes.
